include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -608,17 +618,21 @@ void hemmComputeX(comm::IndexT_MPI reducer_col, matrix::Panel<Coord::Col, T, D>&
   matrix::util::set0<B>(thread_priority::high, x);
   matrix::util::set0<B>(thread_priority::high, xt);
 
-  for (SizeType i = at_offset.rows(); i < dist.localNrTiles().rows(); ++i) {
+  const LocalTileIndex at_offset = view.begin();
+
+  for (SizeType i = at_offset.row(); i < dist.localNrTiles().rows(); ++i) {
     const auto limit = dist.template nextLocalTileFromGlobalTile<Coord::Col>(
         dist.template globalTileFromLocalTile<Coord::Row>(i) + 1);
-    for (SizeType j = limit - 1; j >= at_offset.cols(); --j) {
+    for (SizeType j = limit - 1; j >= at_offset.col(); --j) {
       const LocalTileIndex ij_local{i, j};
       const GlobalTileIndex ij = dist.globalTileIndex(ij_local);
 
       const bool is_diagonal_tile = (ij.row() == ij.col());
 
+      const auto& tile_a = splitTile(a.read(ij), view(ij_local));
+
       if (is_diagonal_tile) {
-        hemmDiag<B>(thread_priority::high, a.read_sender(ij_local), w.read_sender(ij_local),
+        hemmDiag<B>(thread_priority::high, ex::keep_future(tile_a), w.read_sender(ij_local),
```

Comment:
note-to-self: check for `ex::keep_future` and remove it/replace it with `keepFuture`